src/rt/atomic.rs

```diff
@@ -144,24 +144,30 @@ pub(crate) fn fence(order: Ordering) {
 }
 
 impl State {
-    pub(super) fn last_dependent_accesses<'a>(
-        &'a self,
-        action: Action,
-    ) -> Box<dyn Iterator<Item = &'a Access> + 'a> {
-        match action {
-            Action::Load => Box::new({ self.last_load.iter().chain(self.last_store.iter()) }),
-            Action::Store => Box::new({ self.last_load.iter().chain(self.last_store.iter()) }),
-            Action::Rmw => Box::new({ self.last_load.iter().chain(self.last_store.iter()) }),
-        }
+    pub(super) fn for_each_last_dependent_access(
+        &self,
+        _action: Action,
+        mut f: impl FnMut(&Access),
+    ) {
+        self.last_load.iter().for_each(&mut f);
+        self.last_store.iter().for_each(&mut f);
     }
 
-    pub(super) fn set_last_access(&mut self, action: Action, access: Access) {
+    pub(super) fn set_last_access(&mut self, action: Action, path_id: usize, version: &VersionVec) {
+        let set_or_create = |access: &mut Option<Access>| {
```

Comment:
Nit, take it or leave it: would be nice if this repeated code was factored out, maybe a method on `&mut Option<Access>` or something? Not a blocker.

Reply:
If `Access` implemented `clone_from`, it looks like `Option::clone_from` would do something close to what we want out of the box: https://doc.rust-lang.org/src/core/option.rs.html#1191-1196